### DISCUSSION TITLE
Showoff: My Floating desktop

### DISCUSSION THREAD

**Discussion by 24unix** [asmvik/yabai#759] [Show and tell]
I'd like to show you my current setup, and am very interested in seeing yours to get inspirations.

First of all, as the topic suggest, I'm not into tiling. I found yabai because of the transparency, for which I used Deskovery III which stopped working in Catalina.

I'm a switcher from Linux, used KDE since 0.6.7 (back in the 90ies). So I want to make my Macs behave almost like I was used in KDE.

I'm currently achieving this using TotalSpaces2, HyperDock and of course yabai. And a little tool which eventually becomes released under the name Zospeum.

So, I'll try to explain what my main issues were in the beginning.
There are spaces, logical elements.
There are displays, physical elements.
And there are desktops, in my case three displays which should all contain three spaces which belong together.

yabai needs "Displays have separate Spaces" enabled in order to function.

This has pros: The fullscreen mode is usable, without the other display get rendered useless while one app is in fullscreen.
But it also has cons: You cannot force an App to a desktop anymore, only to a screen.

But then I come up with 24 spaces, more than mission control can handle (via shortcut), so I wasn't able anymore to switch my desktops via ctrl-1 to ctrl-8.

I tried out skhd, but it turned to be to slow switching the spaces in sync.

But, I found a way out with TotalSpaces. It can have spaces attached to a display, like so:

<img width="634" alt="Screenshot 2020-12-11 at 15 08 10" src="https://user-images.githubusercontent.com/36622631/101913077-d4834900-3bc2-11eb-940a-84e7afbeda16.png">

So, that is a solution for my desktop problem, displays can have separate spaces but i still have my 8 desktops.

One of the other tools I use is hyperDock, mainly for windows moving. When I hold cmd I can move a windows without clicking the mouse, same for resizing with alt.

<img width="722" alt="hyperdock" src="https://user-images.githubusercontent.com/36622631/101913738-ac481a00-3bc3-11eb-871b-b5fe20bd0439.png">

The last piece in the puzzle (for me) is a tool to access the (for me) most important settings fast from the menu bar.

It currently looks like this:

<img width="868" alt="Zospeum" src="https://user-images.githubusercontent.com/36622631/101914001-09dc6680-3bc4-11eb-90a7-f73408039930.png">

So, maybe you find my current setup interesting. I'm interested in yours, too.

Kind regards,
  Micha

**Comment by 24unix** [comment]
I came a bit further with the small config GUI.
![image](https://user-images.githubusercontent.com/36622631/103137812-b8a49a80-46cc-11eb-9a6b-1cd7ea7a4d4d.png)

**Comment by 24unix** [comment]
So, Total Spaces gave up on Monterey, at least for Intel, TotalSpaces III seems to support AS only.

That lead me to the decision, to handle desktops myself.

Current state:

![image](https://user-images.githubusercontent.com/36622631/145718836-3ab372cd-ac4e-4242-83a2-a44aec879da1.png)

The bar honors Yabais title for a space, and clicking on them switch to current space.
Some glitches still with SwiftUI (ghosting sometimes), but still a good step forward for my workflow.

All of the config stuff from former Zospeum will go into the new App.

**Reply by Ducky6944** [reply]
You might look at [sketchybar](https://github.com/FelixKratz/SketchyBar).

**Comment by 24unix** [comment]
I looked at that, and SwiftBar/BitBar but none of them seems to fit my workflow.
The main thing for me is that all three displays act as one desktop, like it is without Displays have separate Spaces or could be managed with Total Spaces II.

**Comment by 24unix** [comment]
So current state, my App gathers all information regarding display and spaces from yabai:
<img width="1133" alt="image" src="https://user-images.githubusercontent.com/36622631/147574280-297aa91e-ceda-476a-a04b-a173ffaa4aca.png">
And then I can build up my virtual Desktops via Drag’n’Drop:
<img width="645" alt="image" src="https://user-images.githubusercontent.com/36622631/147574451-4f5f8f97-8752-4d2d-8fb4-1f99b86606c8.png">
I’ll switch the layout, exchange X and Y axis, but currently I’m working on syncing the desktops. So when eg space 2 gets activated via skhd, via mouseclick in the dock orby clicking the button in the, also space 11 and 20 need to be activated.

**Comment by rublev** [comment]
Awesome!
